lib/src/simple_config_origin.cc

```diff
@@ -54,6 +57,171 @@ namespace hocon {
         }
     }
 
+    shared_origin simple_config_origin::merge_origins(shared_origin a, shared_origin b) {
+        return merge_two(dynamic_pointer_cast<const simple_config_origin>(a), dynamic_pointer_cast<const simple_config_origin>(b));
+    }
+
+    shared_origin simple_config_origin::merge_origins(std::vector<shared_value> const& stack) {
+        vector<shared_origin> origins;
```

Comment:
It's probably worth doing `origins.reserve(stack.size())`, to ensure we have the right amount of memory and don't have to realloc later.